theia-full-docker/Dockerfile

```diff
@@ -114,11 +114,15 @@ RUN pip install python-language-server
 ##Needed for node-gyp, nsfw build
 RUN apt-get update && apt-get install -y python build-essential
 
+## Needed for JavaScript/TypeScript support
+## See : https://github.com/theia-ide/theia-apps/issues/29
+RUN npm i -g typescript
```

Comment:
I suggest you rebase/squash on top of master, so that we can only see the commit that explicitly add this line instead of the other changes around (which makes it a bit more confusing to review otherwise).